File: Year-2022/LeetCode-August-Challenge/RansomNote.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] count = new int[26];
        for(char ch : magazine.toCharArray()) count[ch - 'a']++;
        for(char ch : ransomNote.toCharArray())
            if(count[ch - 'a']-- <= 0) return false;
        return true;
    }
}
public class RansomNote {
    public static void main(String[] args) {
        Solution s = new Solution();
        String ransomNote = "a", magazine = "b";
        System.out.println(s.canConstruct(ransomNote, magazine));
    }
}

File: Year-2022/LeetCode-August-Challenge/RansomNote.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        int count[26] = {0};
        for(auto i : magazine) count[i - 'a']++;
        for(auto i : ransomNote) 
            if(count[i - 'a']-- <= 0) return false;
        return true;
    }
};
int main(){
    Solution s;
    string ransomNote = "a", magazine = "b";
    cout<<std::boolalpha<<s.canConstruct(ransomNote, magazine)<<endl;
    return 0;
}